### PR TITLE
googleearth-pro: fix xkb keyboard and use nixpkgs libs

### PR DESCRIPTION
**Pull request by SCOTT-HAMILTON**
###### Motivation for this change
Fixes googleearth-pro keyboard not working.

Using Nixpkgs latest Qt libs makes it work way better.

## Security Issues
Among others, googleearth-pro uses : 
 - openssl 1.0.0
 - libav_12
 - icu54

libav_12 still exists on nixpkgs but the two left aren't even there anymore so I had to use the bundled dlls.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

**Pinging Maintainers**
@friedelino

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120449` at db069817 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>googleearth-pro</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/fixup-phase.md">fixup-phase</a>

`fixupPhase` should not be overridden, use `postFixup` instead.

Near pkgs/applications/misc/googleearth-pro/default.nix:101:3:
```
    |
101 |   fixupPhase = ''
    |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/misc/googleearth-pro/default.nix:62:3:
```
   |
62 |   installPhase =''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/misc/googleearth-pro/default.nix [hunk 1]
```diff
@@ -1,41 +1,54 @@
...
-    libxcb
-    sqlite
-    zlib
-    fontconfig
-    libproxy
-    libxml2
-    dbus
-    gst_all_1.gstreamer
-    gst_all_1.gst-plugins-base
-  ];
```
Comment (friedelino) [inline pkgs/applications/misc/googleearth-pro/default.nix:36]: Is this just not the canonical way of adding / pathing the right deps into the binary, or there any other downsides to using `lib.makeLibraryPath` ?

Comment (SCOTT-HAMILTON) [reply pkgs/applications/misc/googleearth-pro/default.nix:36]: I'm not an expert of canonical-derivation making but using the autoPatchelfhook was the easiest way to replace all the bundled dynamic libraries with nixpkgs' one. Now that the cleanup has been done, I could use back makeLibraryPath.

Comment (friedelino) [reply pkgs/applications/misc/googleearth-pro/default.nix:36]: Ah, now I understand. I am also quite a beginner with nix :) So if you don't mind i have another question about the patching when I do `ldd googleearth-bin | grep ssl` now i get something like:
```
libssl.so.1.1 => /nix/store/yc41q33h5xrw1zbyw5hp1y1ga0jk9hwd-openssl-1.1.1k/lib/libssl.so.1.1 (0x00007f0965903000)
libcrypto.so.1.1 => /nix/store/yc41q33h5xrw1zbyw5hp1y1ga0jk9hwd-openssl-1.1.1k/lib/libcrypto.so.1.1 (0x00007f0965616000)
libssl.so.1.0.0 => /nix/store/gxzjr9h9vl2bngswd2w8d540xgmh4y1r-googleearth-pro-7.3.3.7786/opt/google/earth/pro/libssl.so.1.0.0 (0x00007f09618ee000)
```
does this mean, it now also uses the current nixpkg openssl and we could try removing the insecure `libssl-1.0.0` from the bundle?

**Review by mofrim — APPROVED** [review]
Thanks for cleaning this up!

File: pkgs/applications/misc/googleearth-pro/default.nix [hunk 2]
```diff
@@ -89,15 +129,30 @@ stdenv.mkDerivation rec {
...
   '';
 
   # wayland is not supported by Qt included in binary package, so make sure it uses xcb
-  fixupPhase = ''
-    wrapProgram $out/bin/googleearth-pro --set QT_QPA_PLATFORM xcb
+  postFixup = ''
+    wrapProgram $out/bin/googleearth-pro \
+      --set QT_QPA_PLATFORM xcb \
+      --set QT_XKB_CONFIG_ROOT "${xkeyboardconfig}/share/X11/xkb"
   '';
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/googleearth-pro/default.nix:158]: Please update the homepage to https://www.google.com/earth/.

File: pkgs/applications/misc/googleearth-pro/default.nix (hunk 1 above)
Comment (SCOTT-HAMILTON) [reply pkgs/applications/misc/googleearth-pro/default.nix:36]: Nope that's an issue. openssl 1.0.0 is old, unmaintained and has some security issues. Nixpkgs doesn't even have a derivation for it anymore. That's why its using googleearth's one, every dynamic libraries in /nix/store/.....-googleearth-pro/opt/google/earth/pro/ are from the deb file.

Thus the libssl.so.1.0.0 => /nix/store/gxzjr9h9vl2bngswd2w8d540xgmh4y1r-googleearth-pro-7.3.3.7786/**opt/google/earth/pro**/libssl.so.1.0.0 (0x00007f09618ee000)


However you're right on the two others. They're from nixpkgs.

File: pkgs/applications/misc/googleearth-pro/default.nix (hunk 2 above)
Comment (SCOTT-HAMILTON) [reply pkgs/applications/misc/googleearth-pro/default.nix]: Fixed.

**Review by mofrim — COMMENTED** [review]
lgtm

**Comment by SCOTT-HAMILTON** [comment]
Why did you close this PR ?

**Comment by friedelino** [comment]
> Why did you close this PR ?

Closed it by mistake. Sorry for that.

**Review by mofrim — APPROVED** [review]
The proposed changes look good & reasonable to me. We should watch out for new versions of the googleearth binary package which hopefully will drop the insecure deps in the future.

**Comment by SCOTT-HAMILTON** [comment]
> > Why did you close this PR ?
> 
> Closed it by mistake. Sorry for that.

No problem :+1:

**Pull request merged**